kafka/src/main/java/com/expedia/adaptivealerting/kafka/processor/MetricDataTransformer.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2018-2019 Expedia Group, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expedia.adaptivealerting.kafka.processor;
+
+import com.expedia.adaptivealerting.anomdetect.DetectorMapper;
+import com.expedia.adaptivealerting.anomdetect.mapper.Detector;
+import com.expedia.adaptivealerting.anomdetect.mapper.MapperResult;
+import com.expedia.metrics.MetricData;
+import lombok.Data;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.kafka.streams.KeyValue;
+import org.apache.kafka.streams.kstream.Transformer;
+import org.apache.kafka.streams.processor.ProcessorContext;
+import org.apache.kafka.streams.processor.PunctuationType;
+import org.apache.kafka.streams.state.KeyValueIterator;
+import org.apache.kafka.streams.state.KeyValueStore;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+@Slf4j
+@Data
+@RequiredArgsConstructor
+class MetricDataTransformer implements Transformer<String, MetricData, KeyValue<String, MapperResult>> {
```

Comment:
Javadoc please--I don't know what this class is for or why we are adding/removing salt.

Reply:
Now that I better understand this class, I think that this detector lookup logic belongs outside of Kafka. We already talked about it in person, but for the benefit of others who might be following, the thought is that detector lookup is part of the underlying domain logic, and we wouldn't want to lose it if we were to use Kinesis, SNS+AWS lambda, RabbitMQ or some other messaging infrastructure.

In the other processors, we implement the transformers as domain components (see the anomdetect module), and then simply invoke those transformers using `map()`. I'd suggest the same here. I won't treat it as a blocker, but I'd ask that we follow up with this change.